pkg/apis/batch/validation/validation.go

```diff
@@ -665,6 +732,60 @@ func validateCompletions(spec, oldSpec batch.JobSpec, fldPath *field.Path, opts
 	return allErrs
 }
 
+func isConditionTrue(list []batch.JobCondition, cType batch.JobConditionType) bool {
+	if condition := findConditionByType(list, cType); condition != nil {
+		return condition.Status == api.ConditionTrue
+	}
+	return false
+}
+
+func findConditionByType(list []batch.JobCondition, cType batch.JobConditionType) *batch.JobCondition {
+	for i := range list {
+		if list[i].Type == cType {
+			return &list[i]
+		}
+	}
+	return nil
+}
+
+func validateIndexesFormat(indexesStr string, completions int) error {
+	if indexesStr == "" {
+		return nil
+	}
+	var lastIndex *int
+	for _, intervalStr := range strings.Split(indexesStr, ",") {
```

Comment:
I understand this is out of scope in this KEP but please allow me to comment here.

IIUC, we have a similar verification in https://github.com/kubernetes/kubernetes/blob/70af917493983f1d8b1332f8dd0c03800c0a7f15/pkg/controller/job/indexed_job_utils.go#L204.

So, should we commonize these verifications in API pkg, and then can we re-use (import) the verifications in job-controller side?

Reply:
Potentially yes, but I wasn't sure where to put the code. Also, the validation is slightly different than `parseIndexesFromString`, in that function we just log validation failures and move on. Here we return errors, so commonizing would not be straightforward anyway.

Reply:
I see. I'm wondering if we can commonize only processes to parse index, but I think that we can re-visit here another place.

So, tracking this in another issue is better. wdyt?

Reply:
I'm ok as a follow up Issue, but maybe let's create it when this lands. Also, we need to figure out a good package to put such utility functions.